### PR TITLE
Clicking link in alert banner dismisses banner

### PR DESCRIPTION
**Pull request by DamienBurkeWN**
## Testing
* In Storybook you can play with the AlertBanner structure to verify that clicking a link within a banner also dismisses it
* Also fixed the bug whereby a dismissed site alert still flashes up briefly on initial page load in production build; `yalc publish` this branch and on the frontend `yalc add northants-design-system && yarn install && yarn bs`, set a sitewide alert in the CMS, load homepage in frontend, dismiss it, reload page, verify no banner flash.

### CODE REVIEW

File: src/library/structure/AlertBanner/AlertBanner.stories.tsx [hunk 1]
```diff
@@ -28,6 +34,8 @@ const Template: Story<AlertBannerProps> = (args) => (
       <br />
       <br />
       <p>Once this unique alert has been hidden it will not be shown again on this device.</p>
+      <br />
+      <Button url={null} onClick={resetAlert}>Click to reset</Button>
```
Comment (DamienBurkeWN) [inline src/library/structure/AlertBanner/AlertBanner.stories.tsx:38]: Just for our convenience...

File: src/library/structure/AlertBanner/AlertBanner.stories.tsx [hunk 2]
```diff
@@ -20,6 +21,11 @@ const childrenMessage = (
   </p>
```
Comment (chrisrhymes) [inline src/library/structure/AlertBanner/AlertBanner.stories.tsx:21]: Is it worth setting the href="#" so it doesn't redirect to the homepage of storybook?

File: src/library/structure/AlertBanner/AlertBanner.stories.tsx (hunk 1 above)
Comment (chrisrhymes) [reply src/library/structure/AlertBanner/AlertBanner.stories.tsx:38]: Nice!

**Review by chrisrhymes — CHANGES_REQUESTED** [review]
Something is not quite right. If you go on the positive alert example and then dismiss it, then "Click to reset" it is no longer blue, but has a red background? Not sure if it always did that.

File: src/library/structure/AlertBanner/AlertBanner.stories.tsx [hunk 3]
```diff
@@ -16,10 +17,15 @@ export default {
...
 const childrenMessage = (
   <p>
-    Coronavirus | National lockdown: stay at home. <a href="/">Learn what this means for residents and workers here</a>
+    Coronavirus | National lockdown: stay at home. <a href="javascript:;">Learn what this means for residents and workers here</a>
   </p>
 );
 
+const resetAlert = (event) => {
+    window.localStorage.clear();
+    window.location.replace(window.location.href.split('&')[0]);
```
Comment (DamienBurkeWN) [inline src/library/structure/AlertBanner/AlertBanner.stories.tsx:26]: For some reason storybook had rewritten the URL to include empty args, so the straight reload() call ended up with an undefined type for the alert type arg. This strips out any args and I've used replace() so your browser history isn't cluttered with another entry too.

**Review by chrisrhymes — APPROVED** [review]
👍

**Pull request merged**